### PR TITLE
[Tables] updating entities in samples to use more edmtypes

### LINKED ISSUE #18732: [Tables] Update samples to include a wider range of EdmTypes
Currently the samples for creating and updating entities only use string properties in those entities.
We should update the samples to show the full range of EdmTypes in a Python context - both as input and deserialized output.

Specifically these samples:
https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/tables/azure-data-tables/samples/sample_insert_delete_entities.py
https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/tables/azure-data-tables/samples/sample_update_upsert_merge_entities.py

The readme already has a more expansive sample entity - maybe we can leverage this more widely.

### PR DESCRIPTION
**Pull request by seankane-msft**
closes #18732

### CODE REVIEW

File: sdk/tables/azure-data-tables/samples/async_samples/sample_insert_delete_entities_async.py [hunk 1]
```diff
@@ -37,7 +39,17 @@ def __init__(self):
...
+        self.entity = {
+            "PartitionKey": "color",
+            "RowKey": "brand",
+            "text": "Marker",
+            "color": "Purple",
+            "price": 4.99,
+            "last_updated": datetime.today(),
+            "product_id": uuid4(),
+            "inventory_count": 42,
+            "some_binary_value": b"somebinaryvalue"
```
Comment (annatisch) [inline sdk/tables/azure-data-tables/samples/async_samples/sample_insert_delete_entities_async.py:51]: Haha - It looks like the marker merchandise theme petered out a bit :P 
How about:
```python
"barcode": b"somebinaryvalue"
```

**Review by annatisch — COMMENTED** [review]
Thanks for doing this :)

File: sdk/tables/azure-data-tables/samples/async_samples/sample_insert_delete_entities_async.py (hunk 1 above)
Comment (seankane-msft) [reply sdk/tables/azure-data-tables/samples/async_samples/sample_insert_delete_entities_async.py:51]: Haha thank you I could not think of a way to use binary with the marker theme.

**Review by annatisch — APPROVED** [review]

**Pull request merged**